docs/guide/rpc-api.md

```diff
@@ -1,5 +1,309 @@
+Returns a Promise that resolves to an array of a single Ethereum address string.
+If the user denies the request, the Promise will reject with a `4001` error.
+
+The request causes a MetaMask popup to appear.
+You should only request the user's accounts in response to user action, such as a button click.
+Otherwise, you will popup-spam the user.
+
+If you can't retrieve the user's account(s), you should encourage the user to initiate an account request.
+
+#### Example
+
+```javascript
+document.getElementById('connectButton', connect);
+
+function connect() {
+  ethereum
+    .request({ method: 'eth_requestAccounts' })
+    .then(handleAccountsChanged)
+    .catch((error) => {
+      if (error.code === 4001) {
+        // EIP-1193 userRejectedRequest error
+        console.log('Please connect to MetaMask.');
+      } else {
+        console.error(error);
+      }
+    });
+}
+```
+
+### wallet_requestPermissions
+
+```typescript
+interface RequestedPermissions {
+  [methodName: string]: {}; // an empty object, for future extensibility
+}
+
+ethereum.request({
+  method: 'wallet_requestPermissions',
+  params: [ RequestedPermissions ]
+}): Promise<Web3WalletPermission[]>
+```
+
+Requests the given permissions from the user.
+Returns a Promise that resolves to a non-empty array of `Web3WalletPermission` objects, corresponding to the caller's current permissions.
+If the user denies the request, the Promise will reject with a `4001` error.
+
+The request causes a MetaMask popup to appear.
+You should only request permissions in response to user action, such as a button click.
+
+#### Example
+
+```javascript
+document.getElementById('requestPermissionsButton', requestPermissions);
+
+function requestPermissions() {
+  ethereum
+    .request({
+      method: 'eth_requestAccounts',
+      params: [{ eth_accounts: {} }],
+    })
+    .then((permissions) => {
+      const accountsPermission = permissions.find(
+        (permission) => permission.parentCapability === 'eth_accounts'
+      );
+      if (accountsPermission) {
+        console.log('eth_accounts permission successfully requested!');
+      }
+    })
+    .catch((error) => {
+      if (error.code === 4001) {
+        // EIP-1193 userRejectedRequest error
+        console.log('Permissions needed to continue.');
+      } else {
+        console.error(error);
+      }
+    });
+}
+```
+
+### wallet_getPermissions
+
+```typescript
+ethereum.request({
+  method: 'wallet_requestPermissions',
+}): Promise<Web3WalletPermission[]>;
+```
+
+Gets the caller's current permissions.
+Returns a Promise that resolves to an array of `Web3WalletPermission` objects.
+If the caller has no permissions, the array will be empty.
+
+## Other RPC Methods
+
+### wallet_registerOnboarding
+
+::: tip
+As an API consumer, you are unlikely to have to call this method yourself.
+Please see [`@metamask/onboarding`](https://github.com/MetaMask/metamask-onboarding) for more information.
```

Comment:
We should link the docs page about onboarding here instead. The repo README isn't really intended for dapp devs.
```suggestion
Please see the [Onboarding Library documentation](/onboarding-library.html) for more information.
```